src/Symfony/Bridge/PhpUnit/Tests/DeprecationErrorHandler/debug_class_loader_autoload.phpt

```diff
@@ -0,0 +1,47 @@
+--TEST--
+Test that a deprecation from the DebugClassLoader on a vendor class autoload triggered by an app class is considered indirect.
+--FILE--
+<?php
+
+$k = 'SYMFONY_DEPRECATIONS_HELPER';
+putenv($k.'='.$_SERVER[$k] = $_ENV[$k] = 'max[total]=0');
+putenv('ANSICON');
+putenv('ConEmuANSI');
+putenv('TERM');
+
+$vendor = __DIR__;
+while (!file_exists($vendor.'/vendor')) {
+    $vendor = dirname($vendor);
+}
+define('PHPUNIT_COMPOSER_INSTALL', $vendor.'/vendor/autoload.php');
+require PHPUNIT_COMPOSER_INSTALL;
+require_once __DIR__.'/../../bootstrap.php';
+eval(<<<'EOPHP'
+namespace PHPUnit\Util;
+
+class Test
+{
+    public static function getGroups()
+    {
+        return array();
+    }
+}
+EOPHP
+);
+require __DIR__.'/fake_vendor/autoload.php';
+
+// We need the real DebugClassLoader FQCN but in a vendor path.
+$realDebugClassLoaderPath = __DIR__.'/../../../../Component/ErrorHandler/DebugClassLoader.php';
```

Comment:
tests can be run from the root of symfony/symfony, and from the root of the component, that's why tests are failing

Reply:
Should we required-dev symfony/error-handler for this test or just run it in the symfony/symfony context?

Reply:
I suggest to find symfoy root folder, and deduce the path to the file

Reply:
@jderusse in "component" mode, the DebugClassLoader file won't exist because symfony/phpunit-bridge has no symfony/error-handler dependency. I think we just need to skip this test if the DebugClassLoader class does not exist, aren't we?

Reply:
> required-dev symfony/error-handler

do you see any drawback?
if not let's do it